### DISCUSSION TITLE
How to use MapContext {map} methods to respect other components positions?

### DISCUSSION THREAD

**Discussion by molakos** [uber/react-map-gl#1612] [Q&A] [answered]
Hello,

I am struggling to invoke the MapContext's `map.panTo` method. I need to call it outside of my map related component. Upon invocation the animation happens, the map moves to the desired area. The problem is that the `Markers` on the map are remaining on the same positions relative to the map container. It seems like that their absolute position on (x,y) are not modified when the MapContext's map viewport is changed. Plus, if I move the map after the `panTo` method was called, a weird flickering happens and the `Markers` are jumping back on their original position.

Please find an example in this sandbox:
https://codesandbox.io/s/react-mapbox-markers-with-mapcontext-feg9u

Any ideas, solutions would be highly appreciated!

**Comment by Pessimistress** [comment, marked as answer]
https://github.com/visgl/react-map-gl/issues/1294

**Reply by molakos** [reply]
Thank you for the quick reply. I guess I have to pass it down to the map component some other way. I was just hoping for a less edgy solution. :)

**Reply by molakos** [reply]
Hi again,

I actually solved the problem and now I had a little time to create a codepen for others on how to overcome the above mentioned problem. 

Have a look at my codepen: https://codesandbox.io/s/react-mapbox-markers-viewport-updates-from-outside-feg9u

I highly suggest to wrap the Map component's children with React.memo. Be aware that if you will pass down `viewport` into `childrenExtendedProps` then you will have to face a lot of re-renders.

Hope this helps. Cheers
